### ISSUE TITLE
Check: CKV_GCP_76: should check against 3 strings rather than a boolean

### ISSUE THREAD

**Issue by parabolic** [bridgecrewio/checkov#2979]
Hello everyone! I am having an issue with the check `CKV_GCP_76`. I've put all of the details below.

**Describe the issue**
Check: CKV_GCP_76: "Ensure that Private google access is enabled for IPV6" checks for a boolean but eventually it accepts 3 string values.
Those 3 values are:
```
ENABLE_OUTBOUND_VM_ACCESS_TO_GOOGLE
DISABLE_GOOGLE_ACCESS
ENABLE_BIDIRECTIONAL_ACCESS_TO_GOOGLE
```
I've used these commands to find the possible values that this api field accepts.
```sh
gcloud compute networks subnets update --private-ipv6-google-access-type=enable-bidirectional-access <self_link>
gcloud compute networks subnets update --private-ipv6-google-access-type=disable <self_link>
gcloud compute networks subnets update --private-ipv6-google-access-type=enable-outbound-vm-access  <self_link>
```

**Examples**
With the following code I am able to satisfy check `CKV_GCP_76` and `terraform apply` works. 
```tf
resource "google_compute_subnetwork" "subnets" {
....
  private_ipv6_google_access = true
....
}
```
I get a cyclical change on the next plan/apply due to the API not checking the input coming from the SDK and disabling `setPrivateIpv6GoogleAccess` silently.
```sh
      ~ private_ipv6_google_access = "DISABLE_GOOGLE_ACCESS" -> "true"
```
If I change the value in terraform to:
```tf
  private_ipv6_google_access = "ENABLE_OUTBOUND_VM_ACCESS_TO_GOOGLE"
```
Checkov complains with
```sh
Check: CKV_GCP_76: "Ensure that Private google access is enabled for IPV6"
        FAILED for resource: google_compute_subnetwork.subnetwork
....
.....
.....
        24 |   private_ipv6_google_access = "ENABLE_OUTBOUND_VM_ACCESS_TO_GOOGLE"

```
**Version (please complete the following information):**
 - version: 2.0.1140

**Additional context**
I've two paths here, either I disable the check with the `ENABLE_OUTBOUND_VM_ACCESS_TO_GOOGLE` or I leave out `true` and get changes everytime I run terraform.

Thank you!

**Comment by nimrodkor** [comment]
Hey @parabolic !

I agree with your recommendation, and would love to have your contribution to the existing check to align it with what you wrote. You can find it here:
https://github.com/bridgecrewio/checkov/blob/144814d6efd90b399e1cbf37eee953ce392c119f/checkov/terraform/checks/resource/gcp/GoogleSubnetworkIPV6PrivateGoogleEnabled.py

To change the expected value / the value we want to see, you'll need to override the method `get_expected_value`

**Comment by parabolic** [comment]
Hi @nimrodkor I've opened the following PR. https://github.com/bridgecrewio/checkov/pull/2988
Thank you!

**Comment by juangascon** [comment]
@parabolic
Hello. I would like to know where you got the possible values (ENABLE_OUTBOUND_VM_ACCESS_TO_GOOGLE,...) because I do not find them anywhere in internet and I am becoming crazy. 
The only value I get, even if I configure the private IPv6 access in the GCP console, is DISABLE_GOOGLE_ACCESS.
I can not find any of the other two, neither with terraformer nor terraform import.
Luckily, I found this issue.
Thanks in advance.

**Comment by parabolic** [comment]
@juangascon I am glad that this issue has helped you. 
I don't quite remember how I found them out but it was a combination of `gcloud cli` and the GUI.  
Try passing an empty value `gcloud compute networks subnets update --private-ipv6-google-access-type=""` like so and you might get an error. ATM I've no GCP account at my disposal to test and possibly help you.

**Comment by juangascon** [comment]
@parabolic 
Thanks for your answer. Indeed I tried the gcloud command before your answer and I got the API accepted values in the help documentation. 
Though, what I am looking for is the all-capital values you asked to check in this rule:
```
ENABLE_OUTBOUND_VM_ACCESS_TO_GOOGLE
DISABLE_GOOGLE_ACCESS
ENABLE_BIDIRECTIONAL_ACCESS_TO_GOOGLE
```
I mean, the Terraform values accepted for 'private_ipv6_google_access'.
They are the same as those of the API with '_TO_GOOGLE' appended but I could not guess it.
Nevertheless, GCP does not accept the BIDIRECTIONAL parameter and the only one Terraform shows me in the imports is the DISABLE_GOOGLE_ACCESS :-(
Well, if you remember where this is documented, it would be great. If not, it does not matter. I do not want to spend too much time on this, specially when this is not blocking my work.

**Comment by juangascon** [comment]
@parabolic 
Thanks a lot.
WAOUH!! It was on Compute.Instance!!
I was not far on my research. I do not know how I had arrived to the this same page of dotnet documentation but I did not see it. 
I do not understand how I could not see them because searching 'IPv6' in the page, we find the values.
They are even in the Subnetwork section, where I had also looked for.
Pfff, what a moron I am.
Thanks again not only for this info but for showing me the good API reference page.
